app/components/preprint-form-authors.js

```diff
@@ -10,26 +10,158 @@ export default CpPanelBodyComponent.extend(PreprintFormFieldMixin, {
+        let searchResults = this.get('searchResults');
+        let contributors = this.get('contributors');
+        let userIds = contributors.map((contrib) => contrib.id.split('-')[1]);
+        return searchResults.filter((result) => !Ember.A(userIds).contains(result.id));
+    }),
     actions: {
-        addContributor(userId, permission, isBibliographic) {
-            this.sendAction('addContributor', userId, permission, isBibliographic);
+        // Adds contributor then redraws view - addition of contributor may change which update/remove contributor requests are permitted
+        addContributor(user) {
+            this.sendAction('addContributor', user.id, 'write', true);
+            this.toggleProperty('permissionToggle');
+            this.toggleProperty('bibliographicToggle');
+            this.toggleProperty('removalToggle');
+        },
+        // Adds unregistered contributor, then clears form and switches back to search view.
+        // Should wait to transition until request has completed.
+        addUnregisteredContributor(fullName, email) {
+            let res = this.attrs.addUnregisteredContributor(fullName, email, 'write', true);
+            var _this = this;
+            res.then(function() {
+                _this.toggleProperty('bibliographicToggle');
+                _this.toggleProperty('permissionToggle');
+                _this.toggleProperty('removalToggle');
+            });
+            this.set('addState', 'searchView');
+            this.set('fullName', '');
+            this.set('email', '');
+        },
+        updateQuery(value) {
+            this.set('query', value);
         },
+        // Requests a particular page of user results
+        findContributors(page) {
+            var query = this.get('query');
+            var _this = this;
+            if (query) {
+                _this.sendAction('findContributors', query, page);
+                this.set('addState', 'searchView');
+            }
+        },
+        // Removes contributor then redraws contributor list view - removal of contributor may change
+        // which additional update/remove requests are permitted.
         removeContributor(contrib) {
             this.sendAction('removeContributor', contrib);
+            this.toggleProperty('removalToggle');
+            this.toggleProperty('permissionToggle');
+            this.toggleProperty('bibliographicToggle');
+            this.removedSelfAsAdmin(contrib, contrib.get('permission'));
+            this.get('contributors').removeObject(contrib);
         },
-        permissionChange(contributor, permission) {
+        // Updates contributor then redraws contributor list view - updating contributor
+        // permissions may change which additional update/remove requests are permitted.
+        updatePermissions(contributor, permission) {
             this.set(`permissionChanges.${contributor.id}`, permission.toLowerCase());
```

Comment:
Good catch, I don't think this is needed here.  It was being used before when we were doing bulk updates of contributors.